[test][process] delete_mailbox e2e test fails on main; CI excludes e2e so stale failures don't surface

## Problem

\`tests/e2e/test_mcp_tools.py::TestToolInvocation::test_tool_invocation_happy_path[delete_mailbox-...]\` **fails on clean main**:

\`\`\`
WARNING apple_mail_mcp.server:server.py:157 Elicitation unavailable; blocking delete_mailbox: 
        session is not available because the MCP session has not been established yet. 
        Check \`context.request_context\` for None before accessing this attribute.

assert result.structured_content["success"] is True
E   assert False is True
\`\`\`

Surfaced by @allenpan05 during #254 review (they noted "also fails on \`main\`"). I confirmed locally on a clean checkout of main.

## Root cause (test-harness)

The e2e test harness for elicitation-gated tools (like \`delete_mailbox\`) doesn't set up \`ctx.request_context\` cleanly. \`_elicit_confirmation\` in [\`server.py:155\`](src/apple_mail_mcp/server.py#L155) tries to call \`ctx.elicit(summary, None)\` but the underlying MCP session isn't established in the test fixture, so the call fails closed and the tool reports \`success: False\`.

The test was presumably passing at some earlier point; likely broke when the \`_elicit_confirmation\` work landed (#229 / fix-#226 era). The harness needs an updated fixture that provides a working request context or a mock that returns AcceptedElicitation.

## Process root cause (how it was missed)

[\`.github/workflows/test.yml\`](.github/workflows/test.yml#L23) runs:

\`\`\`
uv run pytest tests/ -m "not integration and not e2e and not benchmark"
\`\`\`

**CI explicitly excludes \`e2e\` tests.** \`make check-all\` (the standard local pre-push check) also excludes them — only runs the unit suite. e2e tests are gated behind \`make test-e2e\` with \`MAIL_TEST_MODE=true\` and only get run when someone manually invokes them.

So when an earlier PR broke the elicitation harness for \`delete_mailbox\`:
1. CI was green (excludes e2e)
2. PR merged
3. main is broken on e2e indefinitely until someone happens to run \`make test-e2e\`
4. We have a tested-as-broken state with no automated signal

## Fix scope — two parts

**(a) Fix the test harness** for \`delete_mailbox\` e2e:

- Set up a mock \`ctx.request_context\` in the fixture, or
- Mock \`_elicit_confirmation\` to return success directly
- Add deprecation cleanup while we're in there: \`ctx.elicit(summary, None)\` triggers \`FastMCPDeprecationWarning\` — pass an explicit \`response_type\`

**(b) Close the process gap** so a future stale e2e failure surfaces sooner:

A few options ranked by cost:

- **Cheap:** add e2e to \`make check-all\`. Developers run it locally before pushing, so stale failures bubble up to whoever's about to PR.
- **Medium:** add a CI job that runs e2e tests against a mocked MCP session (no Mail.app needed) — many of the e2e tests are MCP-session-shape tests, not AppleScript tests.
- **Bigger:** isolate "needs Mail.app" e2e tests from "needs only MCP session" e2e tests; CI runs the latter set. Requires categorization pass.

The cheap option is probably enough for v0.9.0 — get a signal in front of contributors during local dev. The medium/bigger versions can be follow-ups.

## Test plan

- After harness fix: \`make test-e2e\` should pass cleanly on a fresh main checkout
- If we move e2e into check-all: \`make check-all\` should be green on main after this lands

## Credit

@allenpan05 for catching the stale failure during their #254 review.

## Related

- #254 (allenpan05's PR — they flagged the failure but their PR doesn't touch it)
- #229 / #226 (elicitation work that probably introduced the regression)

---

## Update: scoping the process fix

Reframed after maintainer feedback: e2e tests will not run in CI by design — they require Mail.app + a test account, which CI can't reliably provide. The realistic policy is **manual e2e runs at two trigger points**:

1. **PRs that touch IMAP or AppleScript code.** Anything modifying `imap_connector.py`, `mail_connector.py` AppleScript bodies, or AppleScript wrappers should run `make test-e2e` locally before pushing. Contributors won't know to do this unless we tell them.
2. **Release prep.** `make test-e2e` becomes a mandatory step before any tagged release. A pre-existing failure on main is a release-blocker, not a known-issue we ship around.

## Revised fix scope

**Tracked here in this issue:**

- **(a) Fix the `delete_mailbox` test harness** — set up `ctx.request_context` cleanly in the fixture so the elicitation gate doesn't fail closed. Also pass an explicit `response_type` to `ctx.elicit()` to clear the `FastMCPDeprecationWarning`.
- **(b) Document the manual e2e policy** in [`CONTRIBUTING.md`](CONTRIBUTING.md) — add a note in the Development Workflow section: "If your PR touches IMAP or AppleScript code paths, run `make test-e2e` before pushing (requires `MAIL_TEST_MODE=true` and a test Mail.app account)."
- **(c) Update the release skill** ([`.claude/skills/release/SKILL.md`](.claude/skills/release/SKILL.md)) — add `make test-e2e` to Phase 9 (Validation) as a mandatory gate, with a note that a pre-existing failure is a release-blocker not a known issue.

The earlier proposal to add e2e to `make check-all` is **withdrawn** — `check-all` is the standard pre-push check, and gating it on Mail.app would block CI runs and headless dev. The trigger-based approach is the right shape.

**Out of scope (not pursued):**

- Mocking the MCP session enough to run e2e in CI without Mail.app — possible but expensive, and the manual policy is cheaper.
- Splitting e2e tests by "needs Mail.app" vs "needs only MCP session" — could enable partial CI later, but not needed now.

--- a/.claude/skills/release/SKILL.md
+++ b/.claude/skills/release/SKILL.md
@@ -49,8 +49,9 @@ Run ALL checks (stop on failure):
 2. `./scripts/check_client_server_parity.sh`
 3. `./scripts/check_complexity.sh`
 4. `make test`
-5. `./scripts/check_dependencies.sh`
-6. `./scripts/check_applescript_safety.sh`
+5. `make test-e2e` — **mandatory** (requires `MAIL_TEST_MODE=true` + a test Mail.app account). CI excludes e2e, so this is the only gate that catches a stale e2e failure. A pre-existing failure on `main` is a **release-blocker**, not a known issue to ship around (#257).
+6. `./scripts/check_dependencies.sh`
+7. `./scripts/check_applescript_safety.sh`
 
 ## Phase 10: Commit, Push, PR
 - Commit: `"release: vX.Y.Z"`

--- a/CONTRIBUTING.md
+++ b/CONTRIBUTING.md
@@ -26,7 +26,8 @@ uv sync --dev
 2. Write tests first (TDD): RED -> GREEN -> REFACTOR
 3. Implement backend (`mail_connector.py`) and frontend (`server.py`) together
 4. Run checks: `make check-all`
-5. Open a PR against `main`
+5. **If your PR touches IMAP or AppleScript code paths** — `imap_connector.py`, `mail_connector.py` AppleScript bodies/wrappers, or tools gated by `_elicit_confirmation` — also run `make test-e2e` before pushing (requires `MAIL_TEST_MODE=true`; the happy-path dispatch tests are mocked and need no account, but the full suite needs a test Mail.app account). **CI does not run e2e** — they need Mail.app — so a stale e2e failure on `main` is only caught by someone running this locally.
+6. Open a PR against `main`
 
 ## Branch Convention
 

--- a/tests/e2e/test_mcp_tools.py
+++ b/tests/e2e/test_mcp_tools.py
@@ -208,6 +208,23 @@ INVOCATION_CASES: list[tuple[str, dict[str, Any], str, Any]] = [
 class TestToolInvocation:
     """Invoke each tool via mcp.call_tool and verify structured response shape."""
 
+    @pytest.fixture(autouse=True)
+    def _accept_elicitation(self, monkeypatch: pytest.MonkeyPatch) -> None:
+        """The e2e harness invokes tools via ``mcp.call_tool`` with no live MCP
+        session, so the real ``_elicit_confirmation`` would fail closed on
+        gated tools (``delete_mailbox``, ``delete_messages``) — its
+        ``ctx.elicit()`` raises "session not available". Simulate user
+        acceptance so the happy-path test exercises tool *dispatch*, not the
+        confirmation flow (which is covered in tests/unit/test_server.py).
+        Class-scoped so it doesn't mask gate behavior elsewhere. (#257)
+        """
+        async def _accept(*_args: object, **_kwargs: object) -> None:
+            return None  # None == user accepted
+
+        monkeypatch.setattr(
+            "apple_mail_mcp.server._elicit_confirmation", _accept
+        )
+
     @pytest.mark.parametrize(
         "tool_name,call_args,connector_method,connector_return",
         INVOCATION_CASES,

